### ISSUE TITLE
Strange Route U-Turn in Middle of Road

### ISSUE THREAD

**Issue by jonfroehlich** [ProjectSidewalk/SidewalkWebpage#352]
I'm guessing this has something to do with the way we use road segments from Open Street Maps and apply them to Google but this was a bit weird:

![image](https://cloud.githubusercontent.com/assets/1621749/19412850/828c73b6-92ec-11e6-90d0-21a896a73c85.png)

**Comment by misaugstad** [comment]
Yeah this is based on how OSM and GSV work. This is almost always the behavior we want, so that one can get a close look at both sides of the street. It isn't always perfect though, as GSV doesn't have images on both sides of the street in some cases (though they seem rare to me).

This seems to be the correct behavior to me, so I'm going to close this issue for now.

**Comment by jonfroehlich** [comment]
I think the more user friendly version of this would have had me turn around at a more intuitive place--like an intersection rather than suddenly in the middle of a street. That was the original inspiration behind my post...

**Comment by misaugstad** [comment]
Ahh I see what you're saying. I'll reopen the issue, as it is an odd user experience. I doubt this will ever be addressed, as the update is probably much too time intensive given the expected improvement in user experience.

**Comment by jonfroehlich** [comment]
Same thing happened here but this time it's a two-sided road. Got to the end of a segment and it wanted me to turn around--but it did show the route on the otherside of the street, I just couldn't easily get to it.

Correct action should be to try and see if there are GSVs on the other side of the road. Otherwise, this is a very confusing "route"

<img width="2530" height="1523" alt="Image" src="https://github.com/user-attachments/assets/942724ec-5061-45cb-a0a2-1b72e5747452" />

**Comment by jonfroehlich** [comment]
Interesting, when I clicked stuck, it did jump me over to the correct road (adjacent).

<img width="2527" height="1498" alt="Image" src="https://github.com/user-attachments/assets/38258a6b-acc9-48a3-99b7-696c5d8b93a1" />

**Comment by misaugstad** [comment]
>Interesting, when I clicked stuck, it did jump me over to the correct road (adjacent).

Yes! The "Go straight" and "Stuck" buttons try to move you to the pano closest to your assigned street. Using the arrows to move to the other side of the street can be more tricky :)
